era5_to_int.py

```
@@ -202,18 +213,50 @@ def add_trailing_slash(str):
         return str + '/'
 
 
+def handle_datetime_args(args):
+    """ Returns starting and ending datetime objects along with an interval
+    timedelta object given an argparse namespace with members 'datetime',
+    'until_datetime', and 'interval_hours'. If 'until_datetime' is None,
+    the returned ending datetime is the same as the starting datetime.
+    If 'interval_hours' is 0, it defaults to a interval of six hours.
+    """
+
+    yyyy, mm, dd, hh = string_to_yyyymmddhh(args.datetime)
+    startDate = datetime.datetime(yyyy, mm, dd, hh)
+
+    if args.until_datetime != None:
+        yyyy, mm, dd, hh = string_to_yyyymmddhh(args.until_datetime)
+        endDate = datetime.datetime(yyyy, mm, dd, hh)
+    else:
+        endDate = startDate
+
+    if args.interval_hours != 0:
+        intvH = datetime.timedelta(hours=args.interval_hours)
+    else:
+        intvH = datetime.timedelta(hours=6)
+
+    return startDate, endDate, intvH
+
+
 if __name__ == '__main__':
     from netCDF4 import Dataset
     import numpy as np
     import WPSUtils
     import argparse
+    import datetime
 
     parser = argparse.ArgumentParser()
     parser.add_argument('datetime', help='the date-time to convert in YYYY-MM-DD_HH format')
+    parser.add_argument('until_datetime', nargs='?', default=None, help='the date-time in YYYY-MM-DD_HH format until which records are converted (Default: datetime)')
+    parser.add_argument('interval_hours', type=int, nargs='?', default=0, help='the interval in hours between records to be converted (Default: 6)')
```

I think this might be better as :
`default=6, help='the interval in hours between records to be converted (Default %(default)s')`

This would allow the above `else` statement in `handle_datetime_args()` to exit or warn the user on negative input rather than silently using the default if a bad input is provided.

---

Good idea. Perhaps now is the time to add other checks in the `handle_datetime_args()` function, for example, checking that `until_datetime` is not before `datetime`.

---

I've pushed a new commit, 0b1ba6e1, to use your suggestion for the default value of `interval_hours`, and also to improve error handling of the positional arguments. I'll plan to squash that commit into bfc7337e before merging this PR.